servers/quarkus-common/src/test/java/org/projectnessie/quarkus/config/TestQuarkusCatalogConfig.java

```diff
@@ -0,0 +1,137 @@
+            .put("nessie.catalog.warehouses.w2.name", "w2")
+            .put("nessie.catalog.warehouses.w2.location", loc12 + "/")
+            .put("nessie.catalog.warehouses.w3.name", "w3")
+            .put("nessie.catalog.warehouses.w3.location", loc3 + "/")
+            .build();
+
+    Map<String, String> cfgWithDefault1 =
+        ImmutableMap.<String, String>builder()
+            .put("nessie.catalog.warehouses.w1.name", "w1")
+            .put("nessie.catalog.warehouses.w1.location", loc12)
+            .put("nessie.catalog.warehouses.w3.name", "w3")
+            .put("nessie.catalog.warehouses.w3.location", loc3)
+            .put("nessie.catalog.default-warehouse.name", "w2")
+            .put("nessie.catalog.default-warehouse.location", loc12)
+            .build();
+
+    Map<String, String> cfgWithDefault2 =
+        ImmutableMap.<String, String>builder()
+            .put("nessie.catalog.warehouses.w3.name", "w3")
+            .put("nessie.catalog.warehouses.w3.location", loc3)
+            .put("nessie.catalog.default-warehouse.name", "w2")
+            .put("nessie.catalog.default-warehouse.location", loc12)
+            .build();
+
+    return Stream.of(
+        arguments(cfgWithoutDefault, "w1", "w1"),
+        arguments(cfgWithoutDefault, loc12, "w1"),
+        arguments(cfgWithoutDefault, loc12 + "/", "w1"),
+        // Actually, the behavior when looking up a warehouse by a location used by multiple
+        // warehouses is undefined, but for this test we can (probably) rely on Immutables.
+        arguments(cfgWithoutDefault, loc12, "w1"),
+        arguments(cfgWithoutDefault, loc12 + "/", "w1"),
+        arguments(cfgWithoutDefault, "w3", "w3"),
+        arguments(cfgWithoutDefault, loc3, "w3"),
+        arguments(cfgWithoutDefault, loc3 + "/", "w3"),
+        //
+        arguments(cfgWithTrailingSlash, "w1", "w1"),
+        arguments(cfgWithTrailingSlash, loc12, "w1"),
+        arguments(cfgWithTrailingSlash, loc12 + "/", "w1"),
+        // Actually, the behavior when looking up a warehouse by a location used by multiple
+        // warehouses is undefined, but for this test we can (probably) rely on Immutables.
+        arguments(cfgWithTrailingSlash, loc12, "w1"),
+        arguments(cfgWithTrailingSlash, loc12 + "/", "w1"),
+        arguments(cfgWithTrailingSlash, "w3", "w3"),
+        arguments(cfgWithTrailingSlash, loc3, "w3"),
+        arguments(cfgWithTrailingSlash, loc3 + "/", "w3"),
+        //
+        arguments(cfgWithDefault1, "w1", "w1"),
+        arguments(cfgWithDefault1, loc12, "w1"),
+        arguments(cfgWithDefault1, loc12 + "/", "w1"),
+        arguments(cfgWithDefault1, loc12, "w1"),
+        arguments(cfgWithDefault1, loc12 + "/", "w1"),
+        arguments(cfgWithDefault1, "w3", "w3"),
+        arguments(cfgWithDefault1, loc3, "w3"),
+        arguments(cfgWithDefault1, loc3 + "/", "w3"),
+        arguments(cfgWithDefault1, "fooooo", "w2"),
```

Comment:
Shouldn't we throw instead? It feels weird to return the default warehouse when the user explicitly asked for some other warehouse (that doesn't exist).